cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -443,6 +443,24 @@ struct AsciiLower {
   }
 };
 
+struct AsciiCapitalizeTransform : public StringTransformBase {
+  int64_t Transform(const uint8_t* input, int64_t input_string_ncodeunits,
+                    uint8_t* output) {
+    if (input_string_ncodeunits > 0) {
+      output[0] = ascii_toupper(input[0]);
+      std::memcpy(output + 1, input + 1, input_string_ncodeunits - 1);
+    }
+    return input_string_ncodeunits;
+  }
+
+  Status InvalidStatus() override {
```

Comment:
It's not clear to me why this function is being overriden if the function always returns successfully.

Reply:
An artifact of copy-paste from another compute function...removed.